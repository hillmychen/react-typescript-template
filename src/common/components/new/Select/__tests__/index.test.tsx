import React from 'react'
import '@testing-library/jest-dom/extend-expect'
import { render } from '@testing-library/react'
import { Select } from '../index'

describe('Select', () => {
  test('test default', () => {
    const { asFragment, getByText } = render(
      <Select
        value={1}
        options={[
          { label: '1', value: 1 },
          { label: '2', value: 2 },
        ]}
      />
    )
    expect(getByText('1')).toHaveClass('ant-select-selection-selected-value')
    expect(asFragment()).toMatchSnapshot()
  })

  test('test undefinedAsAll', () => {
    const { asFragment, getByText } = render(
      <Select
        undefinedAsAll
        open
        options={[
          { label: '1', value: 1 },
          { label: '2', value: 2 },
        ]}
        placeholder="placeholder test"
      />
    )
    expect(getByText('全部')).toBeInTheDocument()
    expect(getByText('placeholder test')).toBeInTheDocument()
    expect(asFragment()).toMatchSnapshot()
  })
})
