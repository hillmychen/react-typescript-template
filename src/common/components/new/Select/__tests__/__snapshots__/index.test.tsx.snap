// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select test default 1`] = `
<DocumentFragment>
  <div
    class="ant-select ant-select-enabled"
  >
    <div
      aria-autocomplete="list"
      aria-controls="test-uuid"
      aria-expanded="false"
      aria-haspopup="true"
      class="ant-select-selection
            ant-select-selection--single"
      role="combobox"
      tabindex="0"
    >
      <div
        class="ant-select-selection__rendered"
      >
        <div
          class="ant-select-selection-selected-value"
          style="display: block; opacity: 1;"
          title="1"
        >
          1
        </div>
      </div>
      <span
        class="ant-select-arrow"
        style="user-select: none;"
        unselectable="on"
      >
        <i
          aria-label="icon: down"
          class="anticon anticon-down ant-select-arrow-icon"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="down"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
            />
          </svg>
        </i>
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Select test undefinedAsAll 1`] = `
<DocumentFragment>
  <div
    class="ant-select ant-select-single ant-select-show-arrow ant-select-open"
  >
    <div
      class="ant-select-selector"
    >
      <span
        class="ant-select-selection-search"
      >
        <input
          aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
          aria-autocomplete="list"
          aria-controls="rc_select_TEST_OR_SSR_list"
          aria-expanded="true"
          aria-haspopup="listbox"
          aria-owns="rc_select_TEST_OR_SSR_list"
          autocomplete="off"
          class="ant-select-selection-search-input"
          id="rc_select_TEST_OR_SSR"
          readonly=""
          role="combobox"
          style="opacity: 0;"
          type="search"
          unselectable="on"
          value=""
        />
      </span>
      <span
        class="ant-select-selection-placeholder"
      >
        placeholder test
      </span>
    </div>
    <span
      aria-hidden="true"
      class="ant-select-arrow"
      style="user-select: none;"
      unselectable="on"
    >
      <span
        aria-label="down"
        class="anticon anticon-down ant-select-suffix"
        role="img"
      >
        <svg
          aria-hidden="true"
          class=""
          data-icon="down"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
          />
        </svg>
      </span>
    </span>
  </div>
</DocumentFragment>
`;
